sale_order_general_discount/models/sale_order_line.py

```diff
@@ -26,8 +20,8 @@ def create(self, vals_list):
 
     @api.depends("order_id", "order_id.general_discount")
     def _compute_discount(self):
-        if hasattr(super(), "_compute_discount"):
-            super()._compute_discount()
+        res = super()._compute_discount()
         for line in self:
-            line.discount = line.order_id.general_discount
-        return
+            if line.order_id.general_discount:
```

Comment:
@jdoutreloux So, if a discount of 0 is set on order level, lines should not be affected ?